### PR TITLE
More encoding

### PR DESCRIPTION
**Pull request by wanddy**
New property "ResponseTextEncoding"  was added to  Web component.If the encoding of web page we request by web component is not UTF8,we cant get the right response content,ResponseTextEncoding property can be set and solve the issue.

### CODE REVIEW

File: appinventor/appengine/src/com/google/appinventor/client/youngandroid/YoungAndroidFormUpgrader.java [hunk 1]
```diff
@@ -1400,6 +1400,15 @@ private static int upgradeWebProperties(Map<String, JSONValue> componentProperti
       // No properties need to be modified to upgrade to version 4.
       srcCompVersion = 4;
     }
+	if (srcCompVersion < 5) {
```
Comment (josmas) [inline appinventor/appengine/src/com/google/appinventor/client/youngandroid/YoungAndroidFormUpgrader.java]: watch out for indentation here; should be 2 spaces for a new line, and 4 if it is a continuation of the previous line.

File: appinventor/components/src/com/google/appinventor/components/runtime/Web.java [hunk 2]
```diff
@@ -226,8 +227,25 @@ public String Url() {
...
+  }
 
   /**
+   * Specifies the Response Text Encoding.
+   */
+  @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_STRING,
+      defaultValue = "UTF-8")
+  @SimpleProperty
+  public void ResponseTextEncoding(String encoding) {
+	  responseTextEncoding = encoding;
```
Comment (josmas) [inline appinventor/components/src/com/google/appinventor/components/runtime/Web.java]: indentation

File: appinventor/components/src/com/google/appinventor/components/runtime/Web.java [hunk 3]
```diff
@@ -989,9 +1007,9 @@ private void processResponseCookies(HttpURLConnection connection) {
     }
   }
 
-  private static String getResponseContent(HttpURLConnection connection) throws IOException {
+  private static String getResponseContent(HttpURLConnection connection,String encoding) throws IOException {
     // Use the content encoding to convert bytes to characters.
-    String encoding = connection.getContentEncoding();
+    //String encoding = connection.getContentEncoding();
```
Comment (josmas) [inline appinventor/components/src/com/google/appinventor/components/runtime/Web.java]: please remove commented out code.

**Comment by wanddy** [comment]
Sorry for my text editer!

**Comment by jisqyv** [comment]
@wanddy Sorry this one has lingered. I expect it to be in the next component release.

**Comment by ewpatton** [comment]
@nicholec You may want to take a look at this as well.

**Comment by ewpatton** [comment]
@jisqyv Are we still committing to have this in the release on Dec 20? Other than a few minor whitespace issues and resolving of two minor conflicts this looks good to me.

**Comment by ewpatton** [comment]
This has been merged as 8c3afbcd1c426fff7fb0d393ee5c665516a3a528

**Pull request closed without merging**